### PR TITLE
✨ Enable Patching of WorkStatus

### LINKED ISSUE #21: bug: cannot patch or update existing WorkStatus
### Describe the bug

Trying to patch/update WorkStatus object metadata with both client and kubectl (patch/edit) produces this error:
```
kubectl --context imbs1 patch workstatuses appsv1-deployment-nginx-nginx-deployment -n cluster1 --type=merge -p '{"metadata":{"annotations":{"version": "v2"}}}'
The WorkStatus "appsv1-deployment-nginx-nginx-deployment" is invalid:
* status.apiVersion: Required value: must not be empty
* status.kind: Required value: must not be empty
```

### Steps To Reproduce

1. create a workstatus
```
kubectl --context imbs1 apply -f - <<EOF
apiVersion: edge.kubestellar.io/v1alpha1
kind: WorkStatus
metadata:
  labels:
    managed-by.kubestellar.io/singletonstatus: "true"
    managed-by.kubestellar.io/wds1.nginx-singleton-bindingpolicy: "true"
  name: test1
  namespace: cluster1
spec:
  sourceRef:
    group: apps
    kind: Deployment
    name: nginx-deployment
    namespace: nginx
    resource: deployments
    version: v1
EOF
```
2. verify that can patch when there is no status set
```
$  kubectl --context imbs1 patch workstatuses test1 -n cluster1 --type=merge -p '{"metadata":{"annotations":{"version": "v2"}}}'
workstatus.edge.kubestellar.io/test1 patched 
```

3. Fill in the status
```
kubectl --context imbs1 -n cluster1 patch workstatus test1 --type='merge' --patch='{"status":{"conditions":[{"type":"Available","status":"True","reason":"DeploymentAvailable","message":"Deployment has minimum availability."}]}}' --subresource=status
```

4. Patch the object:
```
 kubectl --context imbs1 patch workstatuses test1 -n cluster1 --type=merge -p '{"metadata":{"annotations":{"version": "v3"}}}'
The WorkStatus "test1" is invalid: 
* status.apiVersion: Required value: must not be empty
* status.kind: Required value: must not be empty
```

### Expected Behavior

Should be able to patch without validation errors

### Additional Context

It looks like runtime.RawExtension cannot hold arbitrary json content, but it must hold an object that has those object type fields.

This can be verified as follows:

Fill in the status so that it has apiVersion and kind in the object set in the runtime.RawExtension field:
```
kubectl --context imbs1 -n cluster1 patch workstatus test1 --type='merge' --patch='{"status":{"apiVersion":"apps/v1","kind":"Deployment","status":{"availableReplicas":1,"conditions":[{"lastTransitionTime":"2024-02-14T22:10:42Z","lastUpdateTime":"2024-02-14T22:10:42Z","message":"Deployment has minimum availability.","reason":"MinimumReplicasAvailable","status":"True","type":"Available"},{"lastTransitionTime":"2024-02-14T22:10:13Z","lastUpdateTime":"2024-02-14T22:10:42Z","message":"ReplicaSet \"nginx-deployment-7dd74d45b5\" has successfully progressed.","reason":"NewReplicaSetAvailable","status":"True","type":"Progressing"}],"observedGeneration":1,"readyReplicas":1,"replicas":1,"updatedReplicas":1}}}' --subresource=status
workstatus.edge.kubestellar.io/test1 patched
```

try to patch again:
```
$ kubectl --context imbs1 patch workstatuses test1 -n cluster1 --type=merge -p '{"metadata":{"annotations":{"version": "v3"}}}'
workstatus.edge.kubestellar.io/test1 patched
```

### PR DESCRIPTION
**Pull request by waltforme**
## Summary
Resolves #21.

### CODE REVIEW

**Comment by waltforme** [comment]
I hope this could help to unblock https://github.com/kubestellar/kubestellar/issues/1580.

**Review by ezrasilvera — APPROVED** [review]

**Comment by pdettori** [comment]
nice work @waltforme, thank you !!

**Comment by waltforme** [comment]
> nice work @waltforme, thank you !!

YW! @pdettori

File: .github/workflows/workstatus-creation.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: test workstatus creation
+name: test workstatus operations
```
Comment (MikeSpreitzer) [inline .github/workflows/workstatus-creation.yaml:1]: So I would change the name of this file too.

File: api/v1alpha1/workstatus_types.go [hunk 2]
```diff
@@ -48,7 +48,6 @@ type WorkStatusList struct {
 
 // Manifest represents a resource to be deployed
 type RawStatus struct {
-	// +kubebuilder:validation:EmbeddedResource
```
Comment (MikeSpreitzer) [inline api/v1alpha1/workstatus_types.go:51]: This seems like an inferior choice for how to solve the problem.

File: .github/workflows/workstatus-creation.yaml (hunk 1 above)
Comment (waltforme) [reply .github/workflows/workstatus-creation.yaml:1]: Thanks! Did it by #29 .

File: api/v1alpha1/workstatus_types.go (hunk 2 above)
Comment (waltforme) [reply api/v1alpha1/workstatus_types.go:51]: Would be happy to learn about a better solution.

**Pull request merged**